### PR TITLE
chore: remove unused deps and add missing feats for all tomls

### PR DESCRIPTION
**Pull request by wischli**
# Description

Result of fixing `try-runtime` in #1198 required for migration of `pallet-block-rewards`.

* Removes unused deps with help of `cargo-machete` (though this tools creates many false positives, especially for optional deps, `codec` and `scale-info`)
* Adds missing features `std`, `runtime-benchmarks` and `try-runtime` for all crates
  * Enables isolated compiling of any of our crates.
* Adds `try-runtime` to `development-runtime` required in #1198 
* Fixes `try-runtime` which wasn't working since `try-runtime` was made much stricter sometime between Polkadot v0.9.28 and v0.9.32. Basically, each of our pallets requires at least having
```
try-runtime = [ "frame-support/try-runtime" ]
```
  * See [Shawns comment](https://github.com/paritytech/substrate/pull/10174#issuecomment-1234520716) for details.
* Normalizes apostrophes used in tomls to `"` instead of mixed `'`/`"` 
* Adds patch to `PureStake/Frontier`, thanks @NunoAlexandre !

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

```bash
# Find all child directories containing Cargo.toml files
dirs=$(find . -name Cargo.toml -print0 | xargs -0 -n1 dirname | sort -u)

# Execute the command "subalfred check" on each directory
for dir in $dirs; do
    subalfred check features $dir
done

# Check unused deps
# NOTE: creates false positives
cargo-machete
# cargo-machete --fix
```

```
cargo run --release --features=try-runtime try-runtime --chain=development --no-spec-check-panic --execution=native on-runtime-upgrade live --uri=wss://fullnode.parachain.centrifuge.io:443
```

# Checklist:

- [ ] I have added Rust doc comments to structs, enums, traits and functions
- [ ] I have made corresponding changes to the documentation
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I rebased on the latest `main` branch

### CODE REVIEW

**Review by lemunozm — APPROVED** [review]
LGTM! Awesome work @wischli! 🙌🏻

File: pallets/connectors/Cargo.toml [hunk 1]
```diff
@@ -40,7 +40,7 @@ runtime-common = { path = "../../runtime/common", default-features = false }
 # Polkadot
 xcm = { git = "https://github.com/paritytech/polkadot", default-features = false, branch = "release-v0.9.32" }
 
-pallet-ethereum = { git = "https://github.com/PureStake/frontier", default-features = false, branch = "moonbeam-polkadot-v0.9.32" }
```
Comment (mustermeiszer) [inline pallets/connectors/Cargo.toml:43]: Why the change?

File: Cargo.lock [hunk 2]
```diff
@@ -13132,7 +13134,7 @@ version = "1.6.3"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "97fee6b57c6a41524a810daee9286c02d7752c4253064d0b05472833a438f675"
 dependencies = [
- "cfg-if 1.0.0",
+ "cfg-if 0.1.10",
```
Comment (mustermeiszer) [inline Cargo.lock:13137]: Do you know what caused the downgrade of the version?

**Review by mustermeiszer — APPROVED** [review]
Great clean up, love it!! 

Just one minor question regarding a change in the lock.

The other comments are just nit picks that can also be cleaned up on the same run. 

But see no blockers.

File: taplo.toml [hunk 3]
```diff
@@ -22,7 +22,7 @@ column_width = 160
 # Indent based on tables and arrays of tables and their subtables, subtables out of order are not indented.
 indent_tables = false
 # The substring that is used for indentation, should be tabs or spaces (but technically can be anything).
-indent_string = '  '
+indent_string = "  "
```
[the remark this answers is not in the document]
Comment (wischli) [reply taplo.toml:25]: Unfortunately, this does not change anything. There is no `taplo` enforcement on the type of apostrophe used 😓

File: pallets/connectors/Cargo.toml (hunk 1 above)
Comment (wischli) [reply pallets/connectors/Cargo.toml:43]: Unfortunately, I had to point our `pallet-ethereum` dep from the `PureStake/frontier` fork to one newer revision as their `moonbeam-polkadot-v0.9.32` branch is missing one commit for the the `try-runtime` fixes. Now that they have [merged the hotfix](https://github.com/PureStake/frontier/pull/162#event-8657194321) into that branch, I can revert this change.

File: Cargo.lock (hunk 2 above)
Comment (wischli) [reply Cargo.lock:13137]: Very likely due to the patching for `PureStake/frontier`. Should be reverted once I remove the patch.

File: runtime/development/Cargo.toml [hunk 4]
```diff
@@ -304,28 +378,25 @@ try-runtime = [
...
-  "pallet-pool-registry/try-runtime",
+  "pallet-nft/try-runtime",
+  "pallet-offences/try-runtime",
   "pallet-preimage/try-runtime",
   "pallet-proxy/try-runtime",
   "pallet-randomness-collective-flip/try-runtime",
   "pallet-restricted-tokens/try-runtime",
   "pallet-scheduler/try-runtime",
   "pallet-session/try-runtime",
-  "pallet-society/try-runtime",
```
[the remark this answers is not in the document]
Comment (wischli) [reply runtime/development/Cargo.toml:324]: Good point. I will run `cargo-machete` an all crates and remove unused deps!

File: runtime/centrifuge/Cargo.toml [hunk 5]
```diff
@@ -254,9 +260,34 @@ runtime-benchmarks = [
...
   "runtime-common/runtime-benchmarks",
   "xcm-builder/runtime-benchmarks",
+  "cfg-primitives/runtime-benchmarks",
+  "cfg-traits/runtime-benchmarks",
+  "cumulus-pallet-parachain-system/runtime-benchmarks",
+  "cumulus-pallet-xcmp-queue/runtime-benchmarks",
+  "orml-asset-registry/runtime-benchmarks",
+  "orml-tokens/runtime-benchmarks",
+  "orml-xtokens/runtime-benchmarks",
+  "pallet-babe/runtime-benchmarks",
```
[the remark this answers is not in the document]
Comment (wischli) [reply runtime/centrifuge/Cargo.toml]: [d554430](https://github.com/centrifuge/centrifuge-chain/pull/1223/commits/d55443010e333d3141f0670b1ed21799ecd6ec35)

File: runtime/development/Cargo.toml (hunk 4 above)
[the remark this answers is not in the document]
Comment (wischli) [reply runtime/development/Cargo.toml:324]: [d554430](https://github.com/centrifuge/centrifuge-chain/pull/1223/commits/d55443010e333d3141f0670b1ed21799ecd6ec35)

**Review by mustermeiszer — APPROVED** [review]
Thanks for changing Investments cargo already ❤️ 

Reverting patch looks good. Good to go from my side

**Review by NunoAlexandre — APPROVED** [review]
Awesome sauce 🌶️

**Review by NunoAlexandre — APPROVED** [review]
Re-approving

**Comment by mustermeiszer** [comment]
Re-approve

**Review by mustermeiszer — APPROVED** [review]

**Review by cdamian — APPROVED** [review]

**Review by lemunozm — APPROVED** [review]

**Review by offerijns — APPROVED** [review]

**Review by mikiquantum — APPROVED** [review]
